File: tit_for_two_tats.cpp
#include "data_struct.h"

bool tit_for_two_tats(data_struct input_data) {
	bool last_sol = input_data.enemy_solutions.back();
	bool prelast_sol = input_data.enemy_solutions[input_data.enemy_solutions.size() - 2];

	if (!last_sol && !prelast_sol)
		return false;
	else
		return true;
}

File: programs_documentation/tit_for_two_tats/tit_for_two_tats.py
import data_struct

def tit_for_two_tats(input_data: data_struct.data_struct) -> bool:
    if input_data.round_num <= 2:
        return True

    last_sol: bool = input_data.enemy_solutions[-1]
    prelast_sol: bool = input_data.enemy_solutions[-2]

    if not last_sol and not prelast_sol:
        return False

    return True
